### DISCUSSION TITLE
XML payload highlighting

### DISCUSSION THREAD

**Discussion by chris-doehring** [spatie/ray#253] [Ideas]
Hey there,

as there now is support for html and json highlighting, it also would be nice to have XML. Usually XML based API's return a one line XML string which is hard to read. If ray would format and highlight it, that would be a really nice benefit :)

**Comment by chris-doehring** [comment]
To give a practical example: I'm working with the Plesk XML API and usually, their documents look like this:
```xml
<?xml version="1.0" encoding="UTF-8"?>
<packet version="1.6.3.0">[...]</packet>
```

**Comment by patinthehat** [comment]
@chris-doehring I've submitted a PR that adds support for sending formatted XML payloads - spatie/ray#272.  There's not much I can do to highlight the XML, but this should be a good start.

**Reply by chris-doehring** [reply]
Thanks a lot @patinthehat! :) After I created this topic I had some furher thoughts about it and now I fully agree that formatting for XML is more important than highlighting. I would even say that after your PR, further changes regarding highlighting are not really necessary as it will be easy to read anyway.
